src/realm/util/file.cpp

```diff
@@ -1152,6 +1152,11 @@ bool File::rw_lock(bool exclusive, bool non_blocking)
     // or the OS will deadlock when un-suspending the app.
     int mode = exclusive ? O_WRONLY | O_NONBLOCK : O_RDWR | O_NONBLOCK;
 
+    auto report_err = [path = m_fifo_path](int err, const char* msg) {
```

Comment:
Should we worry about the cost of copying this string here?

Reply:
I don't see any reason for this to not just capture `[this]`.

Reply:
👍